### PR TITLE
Spark generic

### PR DESCRIPTION
**Pull request by betobetodev**
Spark-generic plugin allows the utilization of a pool of clusters to submit Spark jobs on the Broker component.

### CODE REVIEW

**Review by igornatanael — APPROVED** [review]
Anybody could also check it here? It looks ok for me

**Review by iurygregory — CHANGES_REQUESTED** [review]
When pep8 is ok we can merge, next commit he is fixing this

File: application_manager/plugins/spark_generic/plugin.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+        for application in self.applications:
+            state = application.get_application_state()
+            if state == "Running":
+                self.masters[application.master] = True
+            else:
+                self.masters[application.master] = False
+
+    def execute(self, data):
+        self.update_running_clusters()
+        if not False in self.masters.values():
```
Comment (armstrongmsg) [inline application_manager/plugins/spark_generic/plugin.py]: maybe this "if" block should be protected by a lock.

File: application_manager/plugins/spark_generic/plugin.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+                                                                master, 
+                                                                local_path))
+
+        spark_job = self._submit_job(master, key_path, main_class, 
+                                     dependencies, local_binary_file, args)
+
+        spark_app_id = spark.get_running_app(master, 
+                                             spark_applications_ids,
+                                             number_of_attempts)
+
```
Comment (armstrongmsg) [inline application_manager/plugins/spark_generic/plugin.py]: Does this plugin use controller or monitor services? Where are the starting calls?

**Review by armstrongmsg — CHANGES_REQUESTED** [review]

File: application_manager/plugins/spark_generic/plugin.py (hunk 2 above)
Comment (betobetodev) [reply application_manager/plugins/spark_generic/plugin.py]: No. We decided this in our last meeting.

File: application_manager/plugins/spark_generic/plugin.py (hunk 1 above)
Comment (betobetodev) [reply application_manager/plugins/spark_generic/plugin.py]: This was fixed by a more pythonic solution.

**Review by iurygregory — APPROVED** [review]

**Review by armstrongmsg — APPROVED** [review]

**Pull request merged**